src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs

```diff
@@ -511,7 +511,9 @@ public class MyClass
     MyClass $$
 }
 ";
-            await VerifyItemExistsAsync(markup, "MyClass", glyph: (int)Glyph.MethodPublic, expectedDescriptionOrNull: CSharpFeaturesResources.Suggested_name);
+            await VerifyItemExistsAsync(markup, "myClass", glyph: (int)Glyph.FieldPublic, expectedDescriptionOrNull: CSharpFeaturesResources.Suggested_name);
+            await VerifyItemExistsAsync(markup, "MyClass", glyph: (int)Glyph.PropertyPublic, expectedDescriptionOrNull: CSharpFeaturesResources.Suggested_name);
+            await VerifyItemExistsAsync(markup, "GetMyClass", glyph: (int)Glyph.MethodPublic, expectedDescriptionOrNull: CSharpFeaturesResources.Suggested_name);
```

Comment:
I have absolutely no idea why this changed... I'll try to investigate but for now I'm changing the test because the new behavior is better than the old one. MyClass has now a property rather than method glyph, which seems more appropriate. GetMyClass still has a method glyph as expected. 

Before:
![image](https://user-images.githubusercontent.com/11444821/38783960-8aa0c6c4-410a-11e8-8716-7cfa76b0775a.png)
After:
![image](https://user-images.githubusercontent.com/11444821/38783961-9543ec0a-410a-11e8-904c-8d6f27775eb4.png)

Reply:
ok with this change?

Reply:
ok turns out this revealed an actual bug... your custom naming styles for methods now don't show up here... this is because a setting for MethodKind.Ordinary now doesn't match for a general SymbolKind.Method and this feature only checks if the style matches generally against a certain symbol kind regardless of the subkind.. I have to fix this :/ and also add tests...

Reply:
The fix involves fixing the name suggestion feature to respect settings for each individual method kind... for example if you have a certain suffix for methods but a different one for local functions, it should respect your style for each one depending on what you're naming....

Reply:
> ok with this change?

Why is the property ordered after the method?  I would prefer we not change the ordering here.  Specifically, having "GetXXX" show up before "XXX" (regardless of the kind) seems less good.

Reply:
hmm, I don't know... The only thing that should really be changed is the icon. I don't know why the order is different.

Reply:
I can't reproduce this anymore. I'm now seeing it in the correct order every time:
![image](https://user-images.githubusercontent.com/11444821/38810911-6ed72b8a-4188-11e8-95b4-e1d468f81f24.png)

Reply:
it seems there's no test helper to assert the order

Reply:
I can't test either the order or the fact that this creates name suggestions based on your preferences, I would have to add so much test infrastructure :-/ I can't even create custom options in the completion tests

Reply:
What I mean is that if I have custom naming styles, let's say properties end with Property, methods end with Method and local functions end with LocalFunction, this behavior should be tested:
![image](https://user-images.githubusercontent.com/11444821/38813250-bccf52c6-418e-11e8-9ede-3caf6728ffd3.png)

Reply:
> it seems there's no test helper to verify the order :/

That seems easy enogh to add :)

Reply:
how? There's about 6 layers between the test method and the actual method that does the checking.. I don't even know where to start or what this API should look like

Reply:
well, just add the write helpers to each layer :)    I mean, that's how we got these tests in the first place :)

This is all code owned by the IDE.  Being able to add a new test method and feed things through is pretty standard when we determine that we need to test something, but don't already ahve the existing infrastructure for it.

Reply:
I'm sorry but the test helpers look like a giant mess, many of the methods have 10 parameters, some of them are even virtual, and they use flags and null to determine what it's going to check in the first place. I don't know how to add anything to that.

Reply:
test for order of suggestions added in bef61dfdec36247c651af31ec87e57bf8f1dcc07 #resolved

Reply:
I went for just creating a new method that returns the completion items so that I can do any kind of regular asserts on them.

Reply:
>I'm sorry but the test helpers look like a giant mess, many of the methods have 10 parameters, some of them are even virtual, and they use flags and null to determine what it's going to actually check in the first place. I don't know how to add anything to that, 

You don't have to add parameters to existing methods.  You can just make a new method.  Something that asserts the order of the completion list.  

> and don't really feel like touching them at all. 

Testing is part of coding.  Just because an existing test system isn't in great shape doesn't mean that testing functionality can be skipped :)

Reply:
the test is added as i said above

Reply:
also this should have been tested before by the person who added the logic to order things

Reply:
Sometimes we're not always lucky enough that the person before did all the things we wished :)

Reply:
Great!